internal/controllers/provisioningrequest_clusterconfig.go

```diff
@@ -167,54 +167,7 @@ func (t *provisioningRequestReconcilerTask) updateConfigurationAppliedStatus(
 		}
 	}()
 
-	// Handle the expected policies path first: if any expected child policies are missing,
-	// gate completion until they appear. Distinguish between:
-	//  - all expected are Inform: Missing (no timeout)
-	//  - otherwise: InProgress or TimedOut
-	if missingChildPolicies {
-		if allExpectedPoliciesInInform {
-			t.object.Status.Extensions.ClusterDetails.NonCompliantAt = nil
-			ctlrutils.SetStatusCondition(&t.object.Status.Conditions,
-				provisioningv1alpha1.PRconditionTypes.ConfigurationApplied,
-				provisioningv1alpha1.CRconditionReasons.Missing,
-				metav1.ConditionTrue,
-				"Not all expected configuration is present",
-			)
-			return
-		}
-
-		// Check if expected policies generation has timed out.
-		policyConfigTimedOut = t.hasPolicyConfigurationTimedOut(ctx)
-		if policyConfigTimedOut {
-			ctlrutils.SetStatusCondition(&t.object.Status.Conditions,
-				provisioningv1alpha1.PRconditionTypes.ConfigurationApplied,
-				provisioningv1alpha1.CRconditionReasons.TimedOut,
-				metav1.ConditionFalse,
-				"Timed out waiting for expected configuration to be present",
-			)
-			if !ctlrutils.IsClusterUpgradeInitiated(t.object) ||
-				ctlrutils.IsClusterUpgradeCompleted(t.object) {
-				ctlrutils.SetProvisioningStateFailed(t.object,
-					"Cluster configuration timed out")
-			}
-		} else {
-			ctlrutils.SetStatusCondition(&t.object.Status.Conditions,
-				provisioningv1alpha1.PRconditionTypes.ConfigurationApplied,
-				provisioningv1alpha1.CRconditionReasons.InProgress,
-				metav1.ConditionFalse,
-				"Expected configuration is not yet prepared",
-			)
-			if !ctlrutils.IsClusterUpgradeInitiated(t.object) ||
-				ctlrutils.IsClusterUpgradeCompleted(t.object) {
-				ctlrutils.SetProvisioningStateInProgress(t.object,
-					"Cluster configuration is being prepared")
-			}
-		}
-		return
-	}
-
-	// No missing expected policies. Keep the existing behavior.
-	if len(targetPolicies) == 0 {
+	if !missingChildPolicies && len(targetPolicies) == 0 {
```

Comment:
nit: it would be beneficial if we can update the function comment to describe the various conditions being updated?